ikarus/utils/tensorutils.hh

```diff
@@ -49,6 +49,22 @@ auto dyadic(const auto& A_ij, const auto& B_kl) {
   return A_ij.contract(B_kl, empty_index_list).eval();
 }
 
+/**
+ * \brief Computes the dyadic product of two first order Tensors (here: Eigen::Vector).
+ * \details The components of the result read  \f[ A_{ij} = a_{i}b_{j}. \f]
+ * \ingroup tensor
+ * \param a_i First tensor.
+ * \param b_j Second tensor.
+ * \return  Resulting tensor after the dyadic product
+ */
+template <typename ST_, int size, bool asTensor = true>
+auto dyadic(const Eigen::Vector<ST_, size>& a, const Eigen::Vector<ST_, size>& b) {
+  if constexpr (asTensor)
+    return tensorView((a * b.transpose()).eval(), std::array<Eigen::Index, 2>({size, size}));
```

Comment:
This should return a tensor and not a tensorview, because the this only works when a and b are not going out of scope and get destroyed.

Reply:
I would just have here `(a * b.transpose()).eval()` and then of somebody needs it use this as `tensorView(dyadic(a,b))`